osquery/config/parsers/file_paths.cpp

```diff
@@ -81,20 +85,48 @@ Status FilePathsConfigParserPlugin::update(const std::string& source,
     data_.add("file_accesses", arr);
   }
 
-  // We know this top-level is an Object.
-  const auto& file_paths = config.at("file_paths").doc();
-  if (file_paths.IsObject()) {
-    for (const auto& category : file_paths.GetObject()) {
-      if (category.value.IsArray()) {
-        for (const auto& path : category.value.GetArray()) {
-          std::string pattern = path.GetString();
-          if (pattern.empty()) {
-            continue;
+  if (config.count("file_paths") > 0) {
+    // We know this top-level is an Object.
+    const auto& file_paths = config.at("file_paths").doc();
+    if (file_paths.IsObject()) {
+      for (const auto& category : file_paths.GetObject()) {
+        if (category.value.IsArray()) {
+          for (const auto& path : category.value.GetArray()) {
+            std::string pattern = path.GetString();
+            if (pattern.empty()) {
+              continue;
+            }
+
+            std::string name = category.name.GetString();
```

Comment:
I think the name of category is the same for every path and it is waste of CPU to create it on each step of the loop. Would you like to create a `name` outside of the loop?